src/mongocxx/options/index.hpp

```diff
@@ -0,0 +1,365 @@
+    /// documents in this collection.
+    ///
+    /// @param expire_after_seconds
+    ///   The amount of time, in seconds, to retain documents.
+    ///
+    /// @see https://docs.mongodb.org/manual/core/index-ttl/
+    ///
+    void expire_after_seconds(std::int32_t expire_after_seconds);
+
+    ///
+    /// The current expire_after_seconds setting.
+    ///
+    /// @return The current expire_after_seconds.
+    ///
+    const stdx::optional<std::int32_t>& expire_after_seconds() const;
+
+    ///
+    /// Sets the index version.
+    ///
+    /// @param v
+    ///   The index version.
+    ///
+    void v(std::int32_t v);
+
+    ///
+    /// The current index version.
+    ///
+    /// @return The current index version.
+    ///
+    const stdx::optional<std::int32_t>& v() const;
+
+    ///
+    /// For text indexes, sets the weight document. The weight document
+    /// contains field and weight pairs.
+    ///
+    /// @param weights
+    ///   The weight document for text indexes.
+    ///
+    void weights(bsoncxx::document::view weights);
+
+    ///
+    /// The current weights setting.
+    ///
+    /// @return The current weights.
+    ///
+    const stdx::optional<bsoncxx::document::view>& weights() const;
+
+    ///
+    /// For text indexes, the language that determines the list of stop words
+    /// and the rules for the stemmer and tokenizer.
+    ///
+    /// @param default_language
+    ///   The default language for text indexes.
+    ///
+    void default_language(std::string default_language);
+
+    ///
+    /// The current default_language setting.
+    ///
+    /// @return The current default_language.
+    ///
+    const stdx::optional<std::string>& default_language() const;
+
+    ///
+    /// For text indexes, the name of the field, in the collection’s documents,
+    /// that contains the override language for the document.
+    ///
+    /// @param language_override
+    ///   The name of the field that contains the override language for text indexes.
+    ///
+    void language_override(std::string language_override);
+
+    ///
+    /// The current language_override setting.
+    ///
+    /// @return The current language_override.
+    ///
+    const stdx::optional<std::string>& language_override() const;
+
+    ///
+    /// Sets the document for the partial filter expression for partial
+    /// indexes.
+    ///
+    /// @param partial_filter_expression
+    ///   The partial filter expression document.
+    ///
+    void partial_filter_expression(bsoncxx::document::view partial_filter_expression);
+
+    ///
+    /// The current partial_filter_expression setting.
+    ///
+    /// @return The current partial_filter_expression.
+    ///
+    const stdx::optional<bsoncxx::document::view>& partial_filter_expression() const;
+
+    ///
+    /// For 2dsphere indexes, the 2dsphere index version number. Version can be
+    /// either 1 or 2.
```

Comment:
Just tried it in the mongo shell and it worked... I have no idea what it does O.O
```
> db.collection.createIndex( { tmp : "2dsphere" }, { "2dsphereIndexVersion": 3 } )
{
	"createdCollectionAutomatically" : true,
	"numIndexesBefore" : 1,
	"numIndexesAfter" : 2,
	"ok" : 1
}
> db.collection.getIndexes()
[
	{
		"v" : 1,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "asdf.collection"
	},
	{
		"v" : 1,
		"key" : {
			"tmp" : "2dsphere"
		},
		"name" : "tmp_2dsphere",
		"ns" : "asdf.collection",
		"2dsphereIndexVersion" : 3
	}
]
> db.colleciton.insert({tmp: [90,90]})
WriteResult({ "nInserted" : 1 })
```

The docs say that we only have two versions:
>MongoDB 2.6 introduces a version 2 of 2dsphere indexes. Version 2 is the default version of 2dsphere indexes created in MongoDB 2.6 and later series. To override the default version 2 and create a version 1 index, include the option { "2dsphereIndexVersion": 1 } when creating the index.

Thoughts??